msw/README.txt

```diff
@@ -142,3 +142,17 @@ Pd font.
 Currently, pdfontloader.dll is pre-built and included within the pdprototype.tgz
 tarball. To build pdfontloader, see https://github.com/pure-data/pdfontloader
 source.
+
+## debug
+
+To avoid stripping symbols out of libtool pre-binaries and debug pd,call configure with --enable-debug and CFLAGS=-g :
```

Comment:
If you manually set `CFLAGS` to `-g`, this will implicitly remove optimization flags. This is typically fine for debug builds, but maybe it would be better to make this explicit, e.g.:

```
If you intend to debug Pd, you probably want an unoptimized build with debug symbols. This can be achieved with the following flags:

./configure --enable-debug CFLAGS="-g -O0"
make
make app
```

(Technically, `CFLAGS=-g` would be sufficient, as it would remove the default CFLAGS, but I don't think that this is obvious.)

Reply:
It's ok for me but when you call it that way, resulting CFLAGS is -O0 -g -O0 since -O0 is prepended in configure script.
It is harmless but redundant ...

As @umlaeute mentions in #1727 -O2 is append when CFLAGS are not set explicitly. This is such an odd behavior to me.
Another option would be to force a CFLAGS reset (with CFLAGS=) and somehow add -g in configure script when --enable-debug is defined, but I think this is discussed in details in #1727 .

Reply:
> -O2 is append when CFLAGS are not set explicitly. This is such an odd behavior to me.

What really happens is that if CFLAGS are not set explicitly, our build system uses the default CFLAGS (that are set by the build environment). On Msys2 this happens to be `-O2`. Now, `--enable-debug` does add `-O0`, but as you note correctly, it is *prepended* to the default CFLAGs, which makes it pretty useless...

But yeah, this is already discussed at length in #1727 and out of scope for this PR. The current behavior is weird anyway, so it probably does not matter whether you suggest `CFLAGS=-g` or `CFLAGS=-g O0`.

---

Side note:

IMO, the underlying problem is that autotools has no notion of build profiles. CMake, on the other hand, provides several useful build profiles out of the box (`Debug`, `Release`, `RelWithDebInfo` and `MinSizeRel`) which just do The Right Thing®.

Reply:
I do like CMake also. Moving to it or any up to date build system could be a great idea.

Reply:
this appears to be very OT for this PR.

Reply:
Sorry, I couldn't resist :-)